Firebase/Auth/Source/Public/FIRAuth.h

```diff
@@ -226,13 +232,21 @@ typedef NS_ENUM(NSInteger, FIRActionCodeOperation) {
 @property(nonatomic, readonly) FIRActionCodeOperation operation;
 
 /** @fn dataForKey:
-    @brief The operation being performed.
+    @brief Deprecated. Please directly use email or previousEmail properties instead.
+ */
+- (NSString *)dataForKey:(FIRActionDataKey)key
+DEPRECATED_MSG_ATTRIBUTE("Please directly use email or previousEmail properties instead.");
 
-    @param key The FIRActionDataKey value used to retrieve the operation data.
+/** @property email
+    @brief The email address to which the code was sent. The new email address in the case of
+        FIRActionCodeOperationRecoverEmail.
+ */
+@property(nonatomic, nullable, readonly, copy) NSString *email;
 
-    @return The operation data pertaining to the provided action code key.
+/** @property previousEmail
+    @brief The current email address in the case of FIRActionCodeOperationRecoverEmail.
```

Comment:
I'm a little confused here - the `previousEmail` property is the "current email address" when recovering? Maybe this just needs a docs change.

Reply:
Yeah, changed to "the email that is being recovered"